tests/test_scheduler.py

```diff
@@ -0,0 +1,110 @@
+        data1 = []
+        for i in range(10):
+            data1.append(ExpInfo("exp" + str(i), {"rid": i, "priority": i}))
+        data2 = []
+        for i in range(100):
+            data1.append(ExpInfo("exp" + str(i), {"rid": i, "priority": 0}))
+        for data in (data1, data2):
+            mdl = scheduler.ExperimentModel()
+            mdl.experimentQueue.extend(data)
+            self.assertEqual(mdl.rowCount(), len(data))
+            for i, exp in enumerate(data):
+                self.assertEqual(mdl.data(mdl.index(i)), exp)
+
+    def test_drag_and_drop(self):
+        mdl = scheduler.ExperimentModel()
+        data = [ExpInfo("exp1", {"rid": 1, "priority": 2}),
+                ExpInfo("exp2", {"rid": 2, "priority": 1}),
+                ExpInfo("exp3", {"rid": 3, "priority": 1})
+               ]
+        mdl.experimentQueue.extend(data)
+        mime0 = QMimeData()
+        mime0.setText("0")
+        mime1 = QMimeData()
+        mime1.setText("1")
+        mime2 = QMimeData()
+        mime2.setText("2")
+        mdl.dropMimeData(mime0, Qt.MoveAction, 0, 0, mdl.index(0)) # exp1 above exp1
+        self.assertEqual(mdl.experimentQueue, data)
+        mdl.dropMimeData(mime0, Qt.MoveAction, 2, 0, mdl.index(0)) # exp1 above exp3
+        self.assertEqual(mdl.experimentQueue, data)
+        mdl.dropMimeData(mime1, Qt.MoveAction, 3, 0, mdl.index(0)) # exp2 below exp3
+        self.assertEqual(mdl.experimentQueue, [data[0], data[2], data[1]])
+        mdl.dropMimeData(mime2, Qt.MoveAction, 1, 0, mdl.index(0)) # exp2 above exp3
+        self.assertEqual(mdl.experimentQueue, data)
+
+
+class TestSchedulerApp(unittest.TestCase):
+    """Unit tests for SchedulerApp class."""
+
+    def setUp(self):
+        self.qapp = QApplication([])
+
+    def tearDown(self):
+        del self.qapp
+
+    def test_add_experiment(self):
+        app = scheduler.SchedulerApp(name="name", parent=QObject())
+        data = [ExpInfo("exp" + str(i), {"rid": i, "priority": 10-i}) for i in range(10)]
+        for info in data:
+            app.addExperiment(info=info)
+        self.assertEqual(app.schedulerFrame.model.experimentQueue, data)
+        app.schedulerFrame.model.experimentQueue = []
+        permutation = [1, 9, 3, 8, 7, 4, 2, 6, 5, 0]
+        inv_permutation = [1, 3, 4, 7, 8, 5, 2, 6, 0, 9]
+        data = [ExpInfo("exp" + str(i), {"rid": i, "priority": permutation[i]}
+                       ) for i in range(10)]
+        for info in data:
+            app.addExperiment(info=info)
+        self.assertEqual(app.schedulerFrame.model.experimentQueue,
+                         [data[inv_permutation[i]] for i in range(10)])
+
+    def test_run_experiment(self):
+        app = scheduler.SchedulerApp(name="name", parent=QObject())
+        experiment_run = ExpInfo("exp1", {"rid": 1, "priority": 1})
+        experiment_queue = ExpInfo("exp2", {"rid": 2, "priority": 2})
+        app.schedulerFrame.model.experimentQueue.append(experiment_run)
```

Comment:
I think you should mock the model here.

What you did is functional test rather than unit test.
We should implement functional tests, too, so please separate it, not removing.